### DISCUSSION TITLE
SMTP server Enable & Disable

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#2803] [Ideas]
Currently there is no way to disable the SMTP server expect for deleting it.  It would be nice to be able disable the SMTP from a site rather than deleting all the settings.

**Comment by W6HBR** [comment]
If changes are being made to the SMTP config, can I ask that we also add a setting to allow sending messages in HTML format?
In the encoding section of NotificationJob.cs, I always add the IsBodyHtml setting.  It would be nice if this was a system setting that could be toggled.

                            // encoding
                            mailMessage.SubjectEncoding = System.Text.Encoding.UTF8;
                            mailMessage.BodyEncoding = System.Text.Encoding.UTF8;
                            mailMessage.IsBodyHtml = true;

**Comment by sbwalker** [comment]
@leigh-pointer I am trying to understand your business scenario... but my guess is that you have an installation with multiple sites and rather than disabling the Notification Job (which would stop sending email for all sites) you would like to disable email delivery for a specific Site. And the only way to accomplish this currently is to remove the SMTP settings for that Site - but then you need to remember to add the settings back later when you want to resume email delivery for that site. Is this correct?

**Reply by leigh-pointer** [reply]
@sbwalker Yes that is correct,

**Comment by sbwalker** [comment]
@W6HBR this would seem to be a Notification property rather than a general SMTP property? ie. if you are creating Notification objects which include HTML content then you would want to flag them appropriately when you create them - and the Notification Job would utilize the property on a per message basis to set IsBodyHtml. Correct?

**Reply by W6HBR** [reply]
@sbwalker Ideally, your approach at the notification level makes sense, but in practical usage, I don't see a difference when using IsBodyHtml for plain text messages.  Setting IsBodyHtml only changes the Content-Type header from "plain/text" to "plain/html", and both messages will still get base64 encoded when sending.  Setting IsBodyHtml is important if you have actual html markup in the message, so the receiving client knows to render it as HTML, but having that set when the content is plain text does not seem to matter.

**Comment by sbwalker** [comment]
@leigh-pointer and @W6HBR #2842 adds both of your enhancement requests

**Reply by leigh-pointer** [reply]
@sbwalker brilliant thank you!
